chimney/src/main/scala/io/scalaland/chimney/internal/compiletime/Exprs.scala

```diff
@@ -0,0 +1,145 @@
+package io.scalaland.chimney.internal.compiletime
+
+import io.scalaland.chimney.partial
+
+private[compiletime] trait Exprs { this: Definitions =>
+
+  /** Platform-specific expression representation (c.universe.Expr[A] in 2, quotes.Expr[A] in 3 */
+  protected type Expr[A]
+  protected object Expr {
```

Comment:
Scope pollution. Previous version with all methods directly under this give you 40+ methods not grouped by any functionality in current scope. And you cannot make it `private` because you have to implement it in a platform-dependent mixin.

Meanwhile moving all extensions to `MethodsImpl` and all implementations to some `CompanionImpl` works relatively nice to API users, it's approach used in Scala 3 `Quotes`.

Reply:
Instead of the following:
```scala
// Expr
object Option {
  def apply[A: Type](a: Expr[A]): Expr[Option[A]] = exprImpl.Option[A](a)
  def empty[A: Type]: Expr[Option[A]] = exprImpl.OptionEmpty[A]
  def apply[A: Type]: Expr[A => Option[A]] = exprImpl.OptionApply[A]
}
// ExprImpl
def Option[A: Type](value: Expr[A]): Expr[Option[A]]
def OptionEmpty[A: Type]: Expr[Option[A]]
def OptionApply[A: Type]: Expr[A => Option[A]]
// implementations of ExprImpl in platform-specific code
```

Why can't you have just the following?
```scala
def Option: OptionModule
trait OptionModule {
  def apply[A: Type](a: Expr[A]): Expr[Option[A]]
  def empty[A: Type]: Expr[Option[A]]
  def apply[A: Type]: Expr[A => Option[A]]
}
// concrete implementation of Option trait in platform-specific code
```

This way you should be able to keep the clean scope (even multiply-nested) and avoid maintaining approximately half of the code of the facade. This should be also closer to what you have in Scala 3 Quotes.

Reply:
Addressed in #293